cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -311,7 +311,7 @@ Result<std::unique_ptr<parquet::arrow::FileReader>> ParquetFileFormat::GetReader
   try {
     reader = parquet::ParquetFileReader::Open(std::move(input), std::move(properties));
   } catch (const ::parquet::ParquetException& e) {
-    return Status::IOError("Could not open parquet input source '", source.path(),
+    return Status::IOError("Could not open Parquet input source '", source.path(),
```

Comment:
I haven't really checked but are we consistent with capitalizing parquet in this way?  It's just a nit but I think I've used lowercase for file types.  E.g. an xml file or a json file.  Although I suppose it's to match the file extension in those cases.

Reply:
I'm not too sure either but I can just revert this, there are also possibly other nits like whether we want to report IPC as 'IPC' or 'Feather'.

I'll also drop the 'Is this a XYZ file', I think it's clear enough with the rest of the context now.